kong/core/plugins_iterator.lua

```diff
@@ -5,6 +5,28 @@ local pl_tablex = require "pl.tablex"
 
 local empty = pl_tablex.readonly {}
 
+local function load_plugin_into_memory(api_id, consumer_id, plugin_name)
+  local rows, err = singletons.dao.plugins:find_all {
+    api_id = api_id,
+    consumer_id = consumer_id,
+    name = plugin_name
+  }
+  if err then
+    return responses.send_HTTP_INTERNAL_SERVER_ERROR(err)
+  end
+
+  if #rows > 0 then
+    for _, row in ipairs(rows) do
+      if api_id == row.api_id and consumer_id == row.consumer_id then
+        return row
+      end
+    end
+  else
+    -- insert a cached value to not trigger too many DB queries.
+    return {null = true}
```

Comment:
The same way we have a `local empty = {}` at the module level here, should we take this as an opportunity to add:

```lua
local t_empty = {}
local t_null = {}
```

?